Arrowgene.Ddon.GameServer/Handler/QuestGetCycleContentsStateListHandler.cs

```diff
@@ -34,14 +35,27 @@ public override void Handle(GameClient client, IPacket packet)
             S2CQuestJoinLobbyQuestInfoNtc pcap = serializer.Read(InGameDump.data_Dump_20A);
             client.Send(pcap);
 #else
-
             S2CQuestJoinLobbyQuestInfoNtc pcap = EntitySerializer.Get<S2CQuestJoinLobbyQuestInfoNtc>().Read(InGameDump.data_Dump_20A);
             S2CQuestJoinLobbyQuestInfoNtc ntc = new S2CQuestJoinLobbyQuestInfoNtc();
 
             ntc.WorldManageQuestOrderList = pcap.WorldManageQuestOrderList; // Recover paths + change vocation
 
             ntc.QuestDefine = pcap.QuestDefine; // Recover quest log data to be able to accept quests
-            ntc.MainQuestIdList = pcap.MainQuestIdList;
+
+            var completedMsq = Server.Database.GetCompletedQuestsByType(client.Character.CommonId, QuestType.Main);
+            foreach (var questId in completedMsq)
+            {
+                ntc.MainQuestIdList.Add(new CDataQuestId() { QuestId = (uint) questId });
+            }
+
+#if false
+            foreach (var quest in QuestManager.GetQuestsByType(QuestType.Tutorial))
+            {
+                ntc.TutorialQuestOrderList.Add(quest.Value.ToCDataTutorialQuestOrderList());
+            }
+#endif
+
+            // pcap.MainQuestIdList;
```

Comment:
So I was considering, should I add this data from the pcap back after we complete all the added quests? That way the content doesn't stay locked and keeps the current stats? Or is restricting this the way you want to go? For example, the BO tree will only have 2 levels available after the Season 1.0 quests. Also, areas outside of Lestania have no information now because we did not progress that far, etc.

Reply:
Merging this PR will mean that it will be possible to add more of the main quest in the future, we want this to go as smoothly as possible. Sure, this means a regression in the BO tree and other stuff but, like many other things, it's the way to go for full restoration.
If we're worried about it we could have a chat command, like /release which unlocks all warp points, that could fill this list

Reply:
Ok, I will leave the commented pcap member just so I can remind myself of it, but will leave the state as matching as what is in the DB. I moved it to being on top of the new code to make it look more associated with that block of code.